coreneuron/io/core2nrn_data_return.cpp

```diff
@@ -538,13 +546,23 @@ void core2nrn_tqueue(NrnThread& nt) {
                     assert(d->type() == SelfEventType);
                     SelfEvent* se = (SelfEvent*) d;
                     int tar_type = se->target_->_type;
-                    int tar_index = se->target_ - nt.pntprocs;
-                    // Note: nt.pntprocs is not permuted.
+                    // Note that instead of getting tar_index from the permuted
+                    // pnt->_i_instance here and for the noweight case above
+                    // which then needs the possibly large inverse permutation
+                    // vectors, it would save some space to use the unpermuted
+                    // nt.pntprocs array along with a much shorter vector
+                    // of type offsets.
```

Comment:
@pramodk Do this now or later?

Reply:
Depends if it's easy to do / quick. Otherwise it would be ok to do a separate/independent PR.

Reply:
I'd prefer to do this in a separate PR as it really does not fix any substantive issues relevant to nrntraub.